builtins/edge/ds/check.go

```diff
@@ -82,6 +85,14 @@ func RegisterCheckRelation(logger *zerolog.Logger, fnName string, dr resolvers.D
 				Trace:    false,
 			})
 			if err != nil {
+				if bctx.TraceEnabled {
```

Comment:
Can we extract this in to helper.go as 

```
func traceError(bctx topdown.BuiltinContext, fnName string, err error) {
	if bctx.TraceEnabled {
		if len(bctx.QueryTracers) > 0 {
			bctx.QueryTracers[0].TraceEvent(topdown.Event{
				Op:      topdown.FailOp,
				Message: fmt.Sprintf("%s error:%s", fnName, err.Error()),
			})
		}
	}
}
```

This will make it reusable and leverage the function name as used in the code at all times.

Reply:
Line 175 there is no trace added for the CheckPermission call